### DISCUSSION TITLE
How do I make the tooltip show up aligned from the middle rather than aligned to the side of the element  when "top-end" or other static position?

### DISCUSSION THREAD

**Discussion by destructo570** [floating-ui/floating-ui#2132] [Q&A] [answered]
![Screenshot 2023-01-21 at 2 09 55 PM](https://user-images.githubusercontent.com/27971772/213859509-07bc3109-51d4-45ee-a957-c209b1b92e16.png)

So in cases the alignment is "top-left" "bottom-end" etc. I want the tooltip to align from the center of the element (And a little bit offset so that the arrow doesn't get out). Right now it starts from the end (placement here is "bottom-end") but I want the right side to start from the red marked position.

How do I achieve this ?

**Comment by atomiks** [comment, marked as answer]
The [`offset#alignmentAxis`](https://floating-ui.com/docs/offset#alignmentaxis) option should do it. 

```js
offset(({rects}) => ({
  alignmentAxis: rects.reference.width / 2 + SOME_STATIC_OFFSET_CONST
});
```

**Reply by destructo570** [reply]
Where to apply this in react ? Sorry I forgot to mention I was using react

**Reply by atomiks** [reply]
```js
useFloating({
  middleware: [
    offset(...)
  ]
});
```

**Reply by destructo570** [reply]
Thank you!
